frontend/src/components/document/TableView/TableView.jsx

```diff
@@ -98,11 +80,42 @@ const TableEntry = (sentence) => {
 }
 
 const TableView = ({ document }) => {
-    const sentences = splitText(document)
-    const tableEntries = sentences.map((sentence) => {
-        return <div key={toText(sentence)}>
-            <TableEntry sentence={sentence} />
-        </div>
+    /*We're going to map each redaction to a tuple (paragraph, indexes)
+    paragraph: A string of the paragraph
+    indexes: (start_index, end_index) of redaction
+    */
+    const paragraph_index_list = document.documentBody.map((paragraph) => {
+        const paragraph_words = paragraph.children.map((child) => child.text).flat(1).join("").split(" ")
+        var curr_word_index = 0
+        const paragraph_index = []
+        paragraph.children.forEach((child) => {
+            if (getRedactionsOnTextNode(child).size > 0) {
+                const index = {
+                    start_index: curr_word_index,
+                    end_index: curr_word_index + child.text.split(" ").length,
+                }
+                paragraph_index.push({
+                    paragraph: paragraph_words,
+                    index: index,
+                })
+            }
+            curr_word_index += child.text.split(" ").length
+        })
+
+        return paragraph_index
+    })
+
+    var count = -1;
+    const tableEntries = []
+    paragraph_index_list.forEach((paragraph_index) => {
```

Comment:
you don't need the manual `count` here, you can get the idx in the `forEach` function
```js
myArray.forEach(function (value, i) {
    console.log('%d: %s', i, value);
});
```